src/services/authServices.ts

```diff
@@ -0,0 +1,20 @@
+import {
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+  signOut,
+} from "firebase/auth";
+import {
+  addDoc,
+  collection,
+  onSnapshot,
+  query,
+  where,
+} from "firebase/firestore";
+import { db } from "../firebaseConfig";
+
+const getUserData = async (id: string) => {
+  const q = query(collection(db, "users"), where("uid", "==", id));
+  onSnapshot(q, (data) => {
```

Comment:
doubt: onSnapshot is a real-time listener. So, are you unsubscribing it anywhere? Because it can consume bandwidth as a background task.

Reply:
Didn't know about it, Thanks Saurabh, I will look into this.